drivers.py

```diff
@@ -92,11 +102,151 @@ def upload_file(self, src, obj_path):
         return dest
 
 
+class GoogleDriveDriver(Driver):
+    """Driver to handle connection to Google Drive"""
+
+    def __init__(self, config):
+        super(GoogleDriveDriver, self).__init__(config)
+
+        try:
+            self._credentials = service_account.Credentials.from_service_account_file(
+                Path(config.google_drive.service_account_file),
+                scopes=["https://www.googleapis.com/auth/drive.file"],
+            )
+
+            self._service: Resource = build(
+                "drive", "v3", credentials=self._credentials
+            )
+
+            self._folder = config.google_drive.folder
+            folder_exists, self._folder_id = self._folder_exists(self._folder)
+
+            if not folder_exists:
+                self._folder_id = self._create_folder(self._folder)
+
+            for email in get_share_with(config.google_drive):
+                if email:
+                    self._share_with(email)
+
+        except Exception as e:
+            raise DriverError("GoogleDrive driver init error: " + str(e))
+
+    def upload_file(self, src: str, obj_path: str):
+        try:
+            file_metadata = {
+                "name": obj_path,
+                "parents": [self._folder_id],
+            }
+            media = MediaFileUpload(src)
+
+            file = (
+                self._service.files()
+                .create(body=file_metadata, media_body=media, fields="id")
+                .execute()
+            )
+
+            file_id = file.get("id")
+
+        except Exception as e:
+            raise DriverError("GoogleDrive driver error: " + str(e))
+
+        return self._file_link(file_id)
+
+    def _folder_exists(self, folder_name: str) -> typing.Tuple[bool, str]:
+        """Check if a folder with the specified name exists. Return boolean and folder ID if exists."""
+
+        # Query to check if a folder with the specified name exists
+        try:
+            query = f"name = '{folder_name}' and mimeType = 'application/vnd.google-apps.folder'"
+            results = (
+                self._service.files().list(q=query, fields="files(id, name)").execute()
+            )
+            items = results.get("files", [])
+        except Exception as e:
+            raise DriverError("Google Drive folder exists error: " + str(e))
+
+        if len(items) > 1:
+            warnings.warn(
```

Comment:
why not using logging - it would be more like logs for users in stdout, no?

Reply:
The whole media sync relies on prints, so introducing logging just here does not seem worth it. I just wanted to have something more specific than print. But I can change this to print like everyting else

Reply:
Well with print we do not control severity.  But ok, let's keep it simple, and use print also here.